File: benchmark/benchmark.js
const fs = require("fs-extra")
const data = fs.readJSONSync("cd4.json");
const TRANS = require("logicle-trans");
const microbenchmark = require("nodemark");

const res = microbenchmark(() => {TRANS.LogicleTransform(data.data, T = 262144, M = 4.5, W = 1, A = 1)});
console.log("Mean runtime:", res.milliseconds(), "milliseconds");

const data_scale = TRANS.LogicleTransform(data.data, 262144, 4.5, 1, 1, false)
console.log(data_scale);
fs.writeJSONSync("cd4-node.json", {data: data_scale}, )

File: benchmark/benchmark.R
library(flowCore)
library(microbenchmark)
library(jsonlite)

fs <- read.FCS("/Volumes/Samsung_T5/FlowJo/LD1_NS+NS_A01_exp.fcs")
logicle.trans <- logicleTransform(w = 1, a = 1, t = 262144, m = 4.5)
hyperlog.trans <- hyperlog()
fast.logicle.trans <- logicleTransform()
cd4 <- fs@exprs[, 12]
jsonlite::write_json(list(data=cd4), "cd4.json")
cd4.scaled.R <- logicle.trans(cd4)

microbenchmark(logicle.trans(cd4), times = 100)


cd4.scaled.node <- jsonlite::fromJSON("cd4-node.json")
sum(abs(cd4.scaled.node$data - cd4.scaled.R/4.5)) / 250342
mean(cd4.scaled.R/4.5)
